contracts/bonding/BondingManager.sol

```diff
@@ -538,6 +597,8 @@ contract BondingManager is ManagerProxyTarget, IBondingManager {
         // Update bonded amount
         del.bondedAmount = currentBondedAmount.add(_amount);
 
+        checkpointDelegator(_owner, del);
```

Comment:
Going further on this and as I did the first tests I came to realize: we should probably ignore the `startRound` for the sake of reward calculation and always grab the `lastClaimRound` instead, or rather just `claimRound`. The `startRound` is actually handled a little differently than the rewards calculation, and to get the historical stake calculated the same way we need to calculate rewards from bonding using the `lastClaimRound` value in the delegator, instead of the `startRound` (which is ignored during rewards calculation).

During an initial bond, the `lastClaimRound` value is set to the current round (not the round+1!) and only the `startRound` is set to round+1. The pending stake is then calculated with `lastClaimRound` [here](https://github.com/livepeer/protocol/blob/f1a6953b2fe952d447eddee0ecd78e8289604f42/contracts/bonding/BondingManager.sol#L1246) in which even though it looks like it considers `lastClaimRound+1`, it actually `.sub(1)`s below). I don't know if this causes any issues in the bigger picture or not, but it was good to learn and it actually simplified this implementation a bit.

The behavior in the code was changed here: https://github.com/livepeer/protocol/pull/611/commits/f1a6953b2fe952d447eddee0ecd78e8289604f42#diff-dccab47588659a417db0f5d573aa0c5c29b97bf949ffbba7a9d825e35f0bfcf7R594